modules/tensor_mechanics/src/materials/StrainEnergyDensity.C

```diff
@@ -26,6 +26,10 @@ StrainEnergyDensityTempl<is_ad>::validParams()
                                "Optional parameter that allows the user to define "
                                "multiple mechanics material systems on the same "
                                "block, i.e. for multiple phases");
+  params.addParam<std::string>("stress_name",
+                               "stress",
+                               "Optional parameter that allows the user to use "
+                               "different stresses on the same material system");
```

Comment:
It may be good to provide additional information since 
"... different stress tensors to compute the strain energy density. For example, it can be used when damage [provide reason]"